kythe/cxx/indexer/cxx/KytheIndexerMain.cc

```diff
@@ -120,9 +120,11 @@ int main(int argc, char* argv[]) {
   options.CreateWorklist = [](IndexerASTVisitor* indexer) {
     return IndexerWorklist::CreateDefaultWorklist(indexer);
   };
+  std::unique_ptr<FileHashRecorder> recorder;
```

Comment:
minor: just use std::optional (or absl::optional); there's no need for heap allocation here.

https://abseil.io/tips/123
https://abseil.io/tips/187